sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -136,19 +142,46 @@ public void addAssignedOverlayPort(long port) {
         return assignedOverlayPorts;
     }
 
-    private static Protos.TaskInfo getTaskInfo(
+    public void setExecutorVolume(VolumeSpec volumeSpec) {
+        // Volumes on the executor must be declared in each TaskInfo.ContainerInfo to be shared among them.
+        for (Protos.TaskInfo.Builder t : getTaskBuilders()) {
+            Protos.ContainerInfo.Builder builder = t.getContainerBuilder();
+
+            if (!builder.hasType()) {
```

Comment:
Should ContainerInfo.Type always be MESOS? If so it may make sense to just always set it here:

```
t.getContainerBuilder()
    .setType(Protos.ContainerInfo.Type.MESOS)
    .addVolumes(volume);
```